server/src/main/java/org/apache/iotdb/db/mpp/execution/schedule/DriverScheduler.java

```diff
@@ -299,13 +299,8 @@ public void abortQuery(QueryId queryId) {
       for (Set<DriverTask> fragmentRelatedTasks : queryRelatedTasks.values()) {
         if (fragmentRelatedTasks != null) {
           for (DriverTask task : fragmentRelatedTasks) {
-            task.lock();
-            try {
-              task.setAbortCause(DriverTaskAbortedException.BY_QUERY_CASCADING_ABORTED);
-              clearDriverTask(task);
-            } finally {
-              task.unlock();
-            }
+            task.setAbortCause(DriverTaskAbortedException.BY_QUERY_CASCADING_ABORTED);
```

Comment:
There is a little problem. If we set abort cause without lock, the origin cause may be occupied.

Reply:
Oh, even though we set it with lock, it can be occupied too. Maybe a status check is necessary, or just delete the abort cause, it  seems we don't care it...

Reply:
> 

Yes, it seems that the last write wins

Reply:
> Oh, even though we set it with lock, it can be occupied too. Maybe a status check is necessary, or just delete the abort cause, it seems we don't care it...

For now we just need it not to be NULL